airflow/contrib/hooks/hdfs_cli_hook.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import logging
+from csv import DictReader
+from datetime import datetime
+from tempfile import NamedTemporaryFile
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base_hook import BaseHook
+from airflow.utils.file import TemporaryDirectory
+from airflow.utils.helpers import popen_and_tail
+
+
+class HDFSCliHook(BaseHook):
+    '''
+    HDFSCliHook is a wrap around "hadoop fs" cli tool
+    '''
+    def __init__(self, hdfs_cli_conn_id='hdfs_cli_default'):
+        self.hdfs_cli_conn_id = hdfs_cli_conn_id
+        self.conn = self.get_connection(hdfs_cli_conn_id)
+        self.binary = self.conn.extra_dejson.get('binary', 'hadoop fs').split(' ')
+        self.fallback_conn_id = \
+            self.conn.extra_dejson.get('fallback_conn_id', 'hdfs_default')
+
+    def run_cli(self, arguments, verbose=True):
+        hdfs_cmd = self.binary + arguments
+        logging.info('HDFS command "%s"', ' '.join(hdfs_cmd))
+        self.sp, tail = popen_and_tail(hdfs_cmd, shell=True)
```

Comment:
I'm not 100% sure about this, but I believe that `shell=True` might make this process ignore termination of the parent process. We don't want that since Airflow uses subprocesses to run/manage tasks. Is it possible to use `shell=False` and maybe something like `shlex.split(hdfs_cmd)` because then you need a list?